lib/rucio/daemons/replicarecoverer/suspicious_replica_recoverer.py

```diff
@@ -338,49 +334,101 @@ def run_once(heartbeat_handler: Any, younger_than: int, nattempts: int, vos: Opt
             # Get the rse_id by going to one of the suspicious replicas from that RSE and reading it from there
             rse_id = list(recoverable_replicas[vo][rse_key].values())[0]['rse_id']
             for replica_key in list(recoverable_replicas[vo][rse_key].keys()):
-                file_metadata = get_metadata(recoverable_replicas[vo][rse_key][replica_key]["scope"], recoverable_replicas[vo][rse_key][replica_key]["name"])
+                file_scope = recoverable_replicas[vo][rse_key][replica_key]["scope"]
+                file_name = recoverable_replicas[vo][rse_key][replica_key]["name"]
+                file_metadata = get_metadata(file_scope, file_name)
                 recoverable_replicas[vo][rse_key][replica_key]["datatype"] = str(file_metadata["datatype"])
                 if recoverable_replicas[vo][rse_key][replica_key]['available_elsewhere'] is True:
                     # Replicas with other copies on at least one other RSE can safely be labeled as bad
                     files_to_be_declared_bad.append(recoverable_replicas[vo][rse_key][replica_key])
                     # Remove replica from dictionary
                     del recoverable_replicas[vo][rse_key][replica_key]
                 elif recoverable_replicas[vo][rse_key][replica_key]['available_elsewhere'] is False:
-                    if (recoverable_replicas[vo][rse_key][replica_key]['name'].startswith("log.")) or (recoverable_replicas[vo][rse_key][replica_key]['name'].startswith("user")):
+                    if (file_name.startswith("log.")) or (file_name.startswith("user")):
                         # Don't keep log files or user files
                         files_to_be_declared_bad.append(recoverable_replicas[vo][rse_key][replica_key])
                         del recoverable_replicas[vo][rse_key][replica_key]
                     else:
                         # Deal with replicas based on their metadata.
                         if file_metadata["datatype"] is None:  # "None" type has no function "split()"
                             files_to_be_ignored.append(recoverable_replicas[vo][rse_key][replica_key])
+                            logger(logging.WARNING, "RSE: %s, replica name %s, surl %s: Replica does not have a data type associated with it. No action will be taken.",
+                                   rse_key, replica_key, recoverable_replicas[vo][rse_key][replica_key]['surl'])
                             continue
+
+                        file_metadata_datatype = file_metadata["datatype"].split("_")[-1]
```

Comment:
In case of `DAOD_PHYS` it would select `PHYS` which is probably not what you wanted to do. Morever `_` in datatype name is very ATLAS specific. I would just keep `datatype` as it is, but rely on regular expression